src/main/scala/com/github/codelionx/distod/actors/Master.scala

```diff
@@ -1,21 +1,118 @@
 package com.github.codelionx.distod.actors
 
-import akka.actor.typed.Behavior
 import akka.actor.typed.scaladsl.{ActorContext, Behaviors}
-import com.github.codelionx.distod.Serialization.CborSerializable
-import com.github.codelionx.distod.actors.Master.Command
+import akka.actor.typed.{ActorRef, Behavior}
+import com.github.codelionx.distod.actors.Master.{Command, LocalPeers}
+import com.github.codelionx.distod.partitions.StrippedPartition
+import com.github.codelionx.distod.protocols.DataLoadingProtocol._
+import com.github.codelionx.distod.protocols.PartitionManagementProtocol
+import com.github.codelionx.distod.protocols.PartitionManagementProtocol._
+import com.github.codelionx.distod.types.{CandidateSet, PartitionedTable}
 
 
 object Master {
 
-  sealed trait Command extends CborSerializable
+  sealed trait Command
+  private final case class WrappedLoadingEvent(dataLoadingEvent: DataLoadingEvent) extends Command
+  private final case class WrappedPartitionEvent(e: PartitionManagementProtocol.PartitionEvent) extends Command
 
-  def apply(): Behavior[Command] = Behaviors.setup( context => new Master(context).start())
+  val name = "master"
+
+  def apply(
+             guardian: ActorRef[LeaderGuardian.Command],
+             dataReader: ActorRef[DataLoadingCommand],
+             partitionManager: ActorRef[PartitionCommand]
+           ): Behavior[Command] = Behaviors.setup(context =>
+    new Master(context, LocalPeers(guardian, dataReader, partitionManager)).start()
+  )
+
+  case class LocalPeers(
+                         guardian: ActorRef[LeaderGuardian.Command],
+                         dataReader: ActorRef[DataLoadingCommand],
+                         partitionManager: ActorRef[PartitionCommand]
+                       )
+  case class CandidateState(
+                             splitCandidates: Seq[Int],
+                             swapCandidates: Seq[(Int, Int)],
+                             isValid: Boolean
+                           )
 }
 
-class Master(context: ActorContext[Command]) {
+class Master(context: ActorContext[Command], localPeers: LocalPeers) {
+
+  import Master._
+  import localPeers._
+
+
+  def start(): Behavior[Command] = initialize()
+
+  private def initialize(): Behavior[Command] = Behaviors.setup { context =>
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method is longer than 50 lines](https://app.codacy.com/app/CodeLionX/distod/pullRequest?prid=4537008)